src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java

```diff
@@ -77,29 +84,21 @@ private String exportToClipboard(Exporter exporter) {
 
             // Write to file:
             exporter.export(panel.getBibDatabaseContext(), tmp,
-                    panel.getBibDatabaseContext()
-                         .getMetaData()
-                         .getEncoding()
-                         .orElse(Globals.prefs.getDefaultEncoding()),
-                    entries);
+                            panel.getBibDatabaseContext()
+                                 .getMetaData()
+                                 .getEncoding()
+                                 .orElse(Globals.prefs.getDefaultEncoding()),
+                            entries);
             // Read the file and put the contents on the clipboard:
-            StringBuilder sb = new StringBuilder();
-            try (Reader reader = new InputStreamReader(Files.newInputStream(tmp, StandardOpenOption.DELETE_ON_CLOSE),
-                    panel.getBibDatabaseContext()
-                         .getMetaData()
-                         .getEncoding()
-                         .orElse(Globals.prefs.getDefaultEncoding()))) {
-                int s;
-                while ((s = reader.read()) != -1) {
-                    sb.append((char) s);
-                }
-            }
-            ClipboardOwner owner = (clipboard, content) -> {
-                // Do nothing
-            };
-            ClipboardContent clipboardContent = new ClipboardContent();
-            clipboardContent.putRtf(sb.toString());
-            Globals.clipboardManager.setContent(clipboardContent);
+
+            String content = readFileToString(tmp);
+
+            DefaultTaskExecutor.runInJavaFXThread(() -> {
```

Comment:
It is slightly better if you return the content here at this point and put it in the clipboard in the `onSuccess` method of the BackgroundTask (which is always executed on the JavaFX thread).